File: backend-common/src/main/java/cc/jktu/apps/common/config/SwaggerConfig.java
package cc.jktu.apps.common.config;


import io.swagger.annotations.Api;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import springfox.documentation.builders.RequestHandlerSelectors;
import springfox.documentation.spi.DocumentationType;
import springfox.documentation.spring.web.plugins.Docket;

@Configuration
public class SwaggerConfig {

    @Bean
    Docket docket(@Value("${spring.application.name}") String name) {
        return new Docket(DocumentationType.OAS_30)
                .select()
                .apis(RequestHandlerSelectors.withClassAnnotation(Api.class))
                .build().pathMapping("/" + name);
    }

}

File: backend-common/src/main/kotlin/cc/jktu/apps/common/config/SwaggerConfig.kt
package cc.jktu.apps.common.config

import io.swagger.v3.oas.models.OpenAPI
import io.swagger.v3.oas.models.info.Info
import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class SwaggerConfig {

    @Bean
    fun openAPI(@Value("\${spring.application.name}") name: String?): OpenAPI {
        val openAPI = OpenAPI()
        openAPI.info(Info().title(name))
        return openAPI
    }
}
